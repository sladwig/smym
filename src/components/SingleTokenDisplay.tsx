import React from 'react';
import './SingleTokenDisplay.css';
// import { ClickPositionEmitter } from './ClickPositionEmitter';

interface IProps {
    value: string;
}

export const SingleTokenDisplay = ({ value }: IProps) => {
    // return <ClickPositionEmitter {...{ value }} />;
};
