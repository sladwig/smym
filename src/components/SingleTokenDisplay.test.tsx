import { render } from '@testing-library/react';
import React from 'react';
import { SingleTokenDisplay } from './SingleTokenDisplay';

describe('SingleTokenDisplay', () => {
    test('renders', () => {
        const { container } = render(<SingleTokenDisplay />);
        expect(container).toMatchSnapshot();
    });

    test('displays with value', () => {
        const { container } = render(<SingleTokenDisplay value="abc" />);
        expect(container.firstChild.innerHtml).toEqual('abc');
    });

    test('can use mappper', () => {
        const mapper = jest.fn().mockReturnValue('333');
        render(<SingleTokenDisplay value="abc" mapper={[mapper, mapper]} />);
        expect(mapper.mock.calls).toEqual([['abc'], ['333']]);
    });
});
