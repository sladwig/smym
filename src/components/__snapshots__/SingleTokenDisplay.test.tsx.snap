// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleTokenDisplay renders 1`] = `<div />`;
